### DISCUSSION TITLE
feat: Progress bar in form

### DISCUSSION THREAD

**Discussion by anirudhaCodes** [charmbracelet/huh#561] [Ideas]
**Is your feature request related to a problem? Please describe.**
It will be challenging to track the progress of the form submission without a visual representation, making it less user-friendly.

**Describe the solution you'd like**
Addition of a progress bar feature for the form will solve this probelm. The progress bar should dynamically update as the user completes each step of the form, providing a visual indication of the overall progress. This progress bar will be optional.

https://github.com/charmbracelet/huh/assets/155895067/3a41b6a4-c6c9-46de-b2c9-55c6f20905de

Assign this to me. (but need better design of the progress bar)

**Comment by maaslalani** [comment]
I quite like this idea, this is something we can definitely add if it's optional!

We'll need to figure out customization and design as well. Nice work @anirudhaCodes!

**Comment by WTIGER001** [comment]
I think that top control is called a "Stepper".. At least that is what Google Material has. I would also like a "vertical" option.  See here for a better description. https://m1.material.io/components/steppers.html

**Comment by bashbunni** [comment]
Hey @anirudhaCodes, thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.
